### PR TITLE
Add support for component data-test-* attributes without values

### PR DESCRIPTION
**Pull request by Turbo87**
This PR is essentially #55 with a small change that restores Ember 1.13 compatibility.

### CODE REVIEW

File: transform-test-selector-params-to-hash-pairs.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* eslint-env node */
+var TEST_SELECTOR_PREFIX = /data-test-.*/;
```
Comment (marcoow) [inline transform-test-selector-params-to-hash-pairs.js]: Can we import this from elsewhere so we don't have to define it at several locations? We also might want to make it configurable in the future.

File: tests/dummy/app/templates/bind-test.hbs [hunk 2]
```diff
@@ -9,3 +9,5 @@
 <div class="test5">{{data-test-component data-test="foo"}}</div>
 
 <div class="test6">{{data-test-component data-non-test="foo"}}</div>
+
+<div class="test7">{{data-test-component data-test-without-value}}</div>
```
Comment (marcoow) [inline tests/dummy/app/templates/bind-test.hbs]: We should have a test for the block form as well:

```hbs
<div class="test8">{{#data-test-component data-test-without-value}}{{/data-test-component}}</div>
```

Comment (pangratz) [inline tests/dummy/app/templates/bind-test.hbs]: Would a test case for `{{component data-test}}` similar to `.test5` we have now be helpful?

File: transform-test-selector-params-to-hash-pairs.js (hunk 1 above)
Comment (pangratz) [inline transform-test-selector-params-to-hash-pairs.js]: We have such a regex already in `addon/utils/bind-data-test-attributes.js`. I think it makes sense to refactor this into a shared function like `isTestSelector`. Would that make sense?

Comment (Turbo87) [inline transform-test-selector-params-to-hash-pairs.js]: yes, but I'd like to do that in a separate PR since we already have quite a bit of duplication around that regex

Comment (Turbo87) [inline transform-test-selector-params-to-hash-pairs.js]: I guess the problem is that one uses ES6 modules and the other CommonJS, so reducing duplication there might not be quite as easy

Comment (marcoow) [inline transform-test-selector-params-to-hash-pairs.js]: 👍

File: tests/dummy/app/templates/bind-test.hbs (hunk 2 above)
Comment (Turbo87) [inline tests/dummy/app/templates/bind-test.hbs]: done

Comment (Turbo87) [inline tests/dummy/app/templates/bind-test.hbs]: done
